src/sagemaker/workflow/emr_step.py

```diff
@@ -61,6 +61,59 @@ def to_request(self) -> RequestType:
         return config
 
 
+def validate_cluster_config(cluster_config, step_name):
+    """Validates user provided cluster_config.
+
+    Args:
+        cluster_config(Union[Dict[str, Any], List[Dict[str, Any]]]):
+            user provided cluster configuration.
+        step_name: The name of the EMR step.
+    """
+
+    instances = "Instances"
+    instancegroups = "InstanceGroups"
+    instancefleets = "InstanceFleets"
+    prefix_with_in = "In EMRStep " + step_name + ", "
```

Comment:
not necessary, or at least define as static instead of local variables.

Reply:
It is necessary. The strings are being used in many places. Wrapping those in a variable would reduce the chance of human error (e.g., typo etc.).

The scope of these variables are limited inside this function. Hence the current local scope is good enough. As an improvement, I've created dedicated error-string templates, and made them static. They are being used from both inside the emr_step.py and test_emr_step.py. It eliminates the redundant code and also reduces the possibility of human error.